HW_14/txt_proxy_writer_reader.py

```diff
@@ -0,0 +1,23 @@
+from txt_reader import TxtReader
+from txt_writer import TxtWriter
+
+
+class TxtProxyWriterReader:
+    def __init__(self, file_path):
+        self.__result = ''
+        self.__is_actual = False
+        self.__txt_reader = TxtReader(file_path)
+        self.__txt_writer = TxtWriter(file_path)
+
+    def read_file(self):
+        if self.__is_actual:
+            return self.__result
+        else:
+            self.__result = self.__txt_reader.read()
+            self.__is_actual = True
+            return self.__result
+
+    def write_to_file(self):
+        self.__is_actual = False
+        self.__txt_writer.write(input())

```

Comment:
where is the proxy? 
You write all the time. 

Let's check if input() == self.__result than do nothing.
In another case write to the file.